core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -979,11 +980,11 @@ private String fullTableName(TableIdentifier ident) {
     return String.format("%s.%s", name(), ident);
   }
 
-  private FileIO newFileIO(SessionContext context, Map<String, String> properties) {
+  protected FileIO newFileIO(SessionContext context, Map<String, String> properties) {
```

Comment:
do we need this scope change for FileIO? Constructor already accepts `ioBuilder` for customization